spec/unit/face/config_spec.rb

```diff
@@ -44,6 +44,53 @@ module PuppetFaceSpecs
     subject.print('all')
   end
 
+  context "when setting config values" do
+    let(:config_file) { '/foo/puppet.conf' }
+    let(:path) { Pathname.new(config_file) }
+    before(:each) do
+      Puppet[:config] = '/foo/puppet.conf'
+      Puppet::FileSystem.stubs(:pathname).with(config_file).returns(path)
+      Puppet::FileSystem.stubs(:touch)
+    end
+
+    it "writes to the correct puppet config file" do
+      Puppet::FileSystem.expects(:open).with(path, anything, anything)
+      subject.set('foo', 'bar')
+    end
+
+    it "creates a config file if one does not exist" do
```

Comment:
Rather than stubbing, I think it would be better to just use the disk to prove file doesn't exist, `set` gets called, file exists.